ansible/modules/hashivault/hashivault_approle_role.py

```diff
@@ -0,0 +1,184 @@
+    token_ttl:
+        description:
+            - Duration to set as the TTL for issued tokens and at renewal time.
+    token_max_ttl:
+        description:
+            - Duration after which the issued token can no longer be renewed.
+    period:
+        description:
+            - Duration of the token generated.
+    enable_local_secret_ids:
+        description:
+            - If set, the secret IDs generated using this role will be cluster local.
+'''
+EXAMPLES = '''
+---
+- hosts: localhost
+  tasks:
+    - hashivault_approle_role:
+        name: ashley
+    - hashivault_approle_role:
+        name: ashley
+        state: absent
+'''
+
+
+def main():
+    argspec = hashivault_argspec()
+    argspec['state'] = dict(required=False, choices=['present', 'absent'], default='present')
+    argspec['name'] = dict(required=True, type='str')
+    argspec['mount_point'] = dict(required=False, type='str', default='approle')
+    argspec['bind_secret_id'] = dict(required=False, type='bool', no_log=True)
+    argspec['bound_cidr_list'] = dict(required=False, type='list')
+    argspec['policies'] = dict(required=False, type='list', default=[])
+    argspec['secret_id_num_uses'] = dict(required=False, type='str')
+    argspec['secret_id_ttl'] = dict(required=False, type='str')
+    argspec['token_num_uses'] = dict(required=False, type='int')
+    argspec['token_ttl'] = dict(required=False, type='str')
+    argspec['token_max_ttl'] = dict(required=False, type='str')
+    argspec['period'] = dict(required=False, type='str')
+    argspec['enable_local_secret_ids'] = dict(required=False, type='bool')
+    module = hashivault_init(argspec)
+    result = hashivault_approle_role(module.params)
+    if result.get('failed'):
+        module.fail_json(**result)
+    else:
+        module.exit_json(**result)
+
+
+@hashiwrapper
+def hashivault_approle_role(params):
+    state = params.get('state')
+    mount_point = params.get('mount_point')
+    name = params.get('name')
+    client = hashivault_auth_client(params)
+    if state == 'present':
+        args = [
+            'policies',
+            'bind_secret_id',
+            'bound_cidr_list',
+            'secret_id_num_uses',
+            'secret_id_ttl',
+            'token_num_uses',
+            'token_ttl',
+            'token_max_ttl',
+            'period',
+            'enable_local_secret_ids',
+        ]
+        desired_state = {}
+        for arg in args:
+            value = params.get(arg)
+            if value is not None:
+                desired_state[arg] = value
+        try:
+            current_state = client.get_role(name, mount_point=mount_point)
+            changed = False
+            missing = []
+            for key in desired_state:
+                if key in current_state:
+                    if current_state[key] != desired_state[key]:
+                        changed = True
+                else:
+                    missing.append(key)
+            if changed:
```

Comment:
you also want to catch for check_mode before actually executing the api call. when check_mode is enabled you dont want to actually execute the api call, just return "changed". ex: `if changed and state == 'present' and not module.check_mode:`